pajbot/bot.py

```
@@ -664,6 +664,32 @@ def timeout_warn(
     def delete_message(self, msg_id: str) -> None:
         self.privmsg(f"/delete {msg_id}")
 
+    def delete_or_timeout(
+        self,
+        user: User,
+        moderation_action: str,
+        msg_id: str,
+        duration: int,
+        reason: Optional[str] = None,
+        disable_warnings: bool = False,
+        once: bool = False,
+    ) -> None:
+        if moderation_action not in ("Delete", "Timeout"):
+            log.warning("moderation_action must only equal Delete or Timeout.")
+            return
+
+        if moderation_action == "Delete" and msg_id is None:
+            log.warning("Cannot use the Delete moderation_action if msg_id is None!!!! BabyRage")
```

lol. But I'd rather you throw an exception here